commit b2bdc45580e04c86571a355b17bf406993eceebc
Author: Mehdi Amini <joker.eph@gmail.com>
Date:   Tue Oct 31 10:29:29 2023

    [mlir][python] Fix possible use of variable use before set
    
    The _mlirRegisterEverything symbol may not be built by some customers. The
    code here was intended to support this, but didn't properly initialize the
    init_module variable.
    This would break JAX with:
    
    NameError: free variable 'init_module' referenced before assignment in enclosing scope

diff --git a/mlir/python/mlir/_mlir_libs/__init__.py b/mlir/python/mlir/_mlir_libs/__init__.py
index 71c074bc955e..d5fc447e49bf 100644
--- a/mlir/python/mlir/_mlir_libs/__init__.py
+++ b/mlir/python/mlir/_mlir_libs/__init__.py
@@ -83,6 +83,7 @@ def _site_initialize():
 
     # If _mlirRegisterEverything is built, then include it as an initializer
     # module.
+    init_module = None
     if process_initializer_module("_mlirRegisterEverything"):
         init_module = importlib.import_module(f"._mlirRegisterEverything", __name__)
 
